viewing_party/party.py

```diff
@@ -1,23 +1,169 @@
 # ------------- WAVE 1 --------------------
 
+import re
+import string
+from xml.dom import IndexSizeErr
+
+
+
 def create_movie(title, genre, rating):
-    pass
+    movies = {}
+
+    if title != None and genre != None and rating != None:
+            movies.update({"title": title })
+            movies.update({"genre": genre})
+            movies.update({"rating": float(rating) })
+    else:
+        return None
+    
+    return movies
+
+def add_to_watched(user_data, movies):
+    
+    user_data = {
+        "watched" : [movies]
+    }
+    return user_data
+
+
+def add_to_watchlist(user_data, movies):
+    user_data = {
+        "watchlist" : [movies]
+    }
+    return user_data
+
+#adds movie to watched list
+def watch_movie (user_data, movie_to_watch):
+    
+        for movie in user_data["watchlist"]:
+            if movie["title"] == movie_to_watch:
+                user_data["watched"].append(movie)
+                user_data["watchlist"].remove(movie)
+        return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    total_ratings = []
+    if user_data["watched"] != []:
+        try:
+            for movie in user_data["watched"]:
+                total_ratings.append(movie['rating'])
+
+            number_of_ratings = len(total_ratings)
+            average_rating = sum(total_ratings)/int(number_of_ratings)
+        except ZeroDivisionError:
+            average_rating = 0
+    else:
+        average_rating = 0
+
+    return average_rating
 
+#genre is a string
+def get_most_watched_genre(user_data):
+
+    if user_data["watched"] != []:
+        try:
+
+            most_watched = {}
+            description = []
+            for movie in user_data["watched"]:
+                if movie['genre']:
+                    description.append(movie['genre'])
+            for genre in description:
+                if genre in most_watched:
+                    most_watched[genre] += 1
+                else:
+                    most_watched[genre] = 1
+            most_popular = max(most_watched.values())
+            for keys, values in most_watched.items():
+                if values == most_popular:
+                    return keys
+                return user_data
+        except None:
+            return  None
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def create_friends_dictionary(user_data):
+    friends_movies = {}
+    friends_movies[1] = user_data["friends"][0]
+    friends_movies[2] = user_data["friends"][1]

```

Comment:
What would happen here if the friends list in `user_data` _didn't_ have at least 2 friends? What would happen if there were _more_ than 2 friends?

We should avoid making assumptions about how many records there will be in general, even though all the tests in wave 3 (and it turns out beyond) happen to have 2 friends.